src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -890,13 +919,14 @@ private void ExecuteBlockActions<TBlockAction>(
                 var codeBlockAction = blockAction as CodeBlockAnalyzerAction;
                 if (codeBlockAction != null)
                 {
+                    // This context is a struct, so it's fine to create a new one for each action.
                     var context = new CodeBlockAnalysisContext(declaredNode, declaredSymbol, semanticModel,
                         AnalyzerOptions, addDiagnostic, isSupportedDiagnostic, filterSpan, isGeneratedCode, cancellationToken);
 
                     ExecuteAndCatchIfThrows(
                         codeBlockAction.Analyzer,
-                        data => data.action(data.context),
-                        (action: codeBlockAction.Action, context: context),
+                        static data => data.codeBlockAction.Action(data.context),
+                        (codeBlockAction, context),
                         new AnalysisContextInfo(Compilation, declaredSymbol, declaredNode),
```

Comment:
same with this.  the moving to strongly typed code will allow us to live this out of the loop.